starlette/datastructures.py

```diff
@@ -266,33 +290,43 @@ def __init__(
         self._dict = {k: v for k, v in _items}
         self._list = _items
 
-    def getlist(self, key: typing.Any) -> typing.List[typing.Any]:
+    def getlist(self, key: typing.Any) -> typing.List[_VT_co]:
         return [item_value for item_key, item_value in self._list if item_key == key]
 
-    def keys(self) -> typing.KeysView:
+    def keys(self) -> typing.KeysView[_KT]:
         return self._dict.keys()
 
-    def values(self) -> typing.ValuesView:
+    def values(self) -> typing.ValuesView[_VT_co]:
         return self._dict.values()
 
-    def items(self) -> typing.ItemsView:
+    def items(self) -> typing.ItemsView[_KT, _VT_co]:
         return self._dict.items()
 
-    def multi_items(self) -> typing.List[typing.Tuple[str, str]]:
+    def multi_items(self) -> typing.List[typing.Tuple[_KT, _VT_co]]:
         return list(self._list)
 
-    def get(self, key: typing.Any, default: typing.Any = None) -> typing.Any:
+    @typing.overload
+    def get(self, key: _KT) -> _VT_co:
+        ...  # pragma: no cover
+
+    @typing.overload
+    def get(
+        self, key: _KT, default: typing.Optional[typing.Union[_VT_co, _T]] = ...
+    ) -> typing.Union[_VT_co, _T]:
+        ...  # pragma: no cover
+
+    def get(self, key: _KT, default: typing.Any = None) -> typing.Any:
         if key in self._dict:
-            return self._dict[key]
+            return typing.cast(_VT_co, self._dict[key])
         return default
```

Comment:
Not sure if I fully understand the reason of what is written here. :thinking:

Reply:
This is saying:

- If you do not provide a default, the return value is going to be of type `_VT_co` (so a value of the dictionary) (or an error)
- If you do provide a default value, the return value is going to be either `_VT_co` (if the key is found) or the type of your default value (if the key is not found)

Reply:
I removed the `typing.cast` by giving `ImmutableMultiDict._dict` a type annotation